chsdi/tests/integration/test_catalog.py

```diff
@@ -28,3 +28,62 @@ def test_catalog_ordering(self):
         self.failUnless(resp.content_type == 'application/json')
         self.failUnless('AGNES' in resp.json['results']['root']['children'][0]['children'][0]['children'][0]['label'])
         self.failUnless('Geoid in CH1903' in resp.json['results']['root']['children'][0]['children'][0]['children'][1]['label'])
+
+    def test_catalog_languages(self):
+        for lang in ('de', 'fr', 'it', 'rm', 'en'):
+            link = '/rest/services/ech/CatalogServer?lang=' + lang
+            resp = self.testapp.get(link)
+            self.failUnless(resp.status_int == 200, link)
+
+    def test_all_catalogs(self):
+
+        def existInTree(id, root):
+            if (root['id'] == id):
+                return True
+            if 'children' in root:
+                for child in root.get('children'):
+                    if (existInTree(id, child)):
+                        return True
+            return False
+
+        def existInList(node, l):
+            found = False
+            for entry in l:
+                if entry.id == node.get('id'):
+                    found = True
+                    break
+
+            if not found:
+                print node.get('id')
```

Comment:
Yes, we have traceback, but we only know the line and since this is in a loop, we don't know which ID fails, therefore we print it to have it in console.